metasync.js

```diff
@@ -72,3 +72,28 @@ metasync.sequential = function(funcs, done) {
   if (len > 0) next();
   else done();
 };
+
+
+// Asynchrous filter
+// filter :: (a -> Boolean) -> [a] -> ([a] -> Void)
+metasync.filter = function(predicate, array, done) {
+  var result = [];
+  var index = 0;
+  
+  function doFilter() {
+    if (index === array.length) {
+      done(result);
+    } else {
+      if (predicate(array[index])) {
```

Comment:
How is that supposed to look when the function is used?  The only way I see it, is the following one: 
```js
function predicate(item, callback) {
  /* some precalculations */
  callback(/* some condition on item */);
}

metasync.filter(predicate, [/* some data */], done);
```
But why should developer create such monstrous predicate functions? What is the profit he gets?